### PR TITLE
Add variables to road sign concepts

### PR DESCRIPTION
**Pull request by andreo141**
## Overview
Added a component to the road sign concept form to add variables specific to a road sign.

These variables are not used in a template, but rather are used to describe certain aspects of the sign. E.g. how many traffic lines (number), …

##### connected issues and PRs:
https://binnenland.atlassian.net/browse/GN-4924

app: https://github.com/lblod/app-mow-registry/pull/93

### Setup

use branch `feature/add-variables-to-signs` in app

### How to test/reproduce
Go to road-sign-concepts module and edit/create a road sign concept. The form will have a new component that will let you add variables.

### CODE REVIEW

File: app/templates/road-sign-concepts/road-sign-concept.hbs [hunk 1]
```diff
@@ -97,10 +97,28 @@
...
+          </dt>
+          <dd class="au-u-muted">
+            {{#if @model.roadSignConcept.variables}}
+              <ul>
+                {{#each @model.roadSignConcept.variables as |variable|}}
+                  <li>
+                    {{variable.value}}
+                  </li>
+                {{/each}}
+              </ul>
```
Comment (Windvis) [inline app/templates/road-sign-concepts/road-sign-concept.hbs:113]: I would add an `{{else}}` here, similar to the shapes. Or alternatively hide the whole <div> so the title isn't shown either.

File: app/components/road-sign-form/variable-manager.hbs [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+          as |type|
+        >
+          {{type.label}}
+        </PowerSelect>
+        <AuButton @skin="secondary" {{on "click" this.addVariable}}>{{t
+            "utility.add"
+          }}</AuButton>
+      </AuButtonGroup>
+
+      <AuDataTable
```
Comment (Windvis) [inline app/components/road-sign-form/variable-manager.hbs:32]: I would use an `AuTable`

**Review by Windvis — COMMENTED** [review]
Some early feedback, I only looked at the code for now. I'll test it later.

File: app/templates/road-sign-concepts/road-sign-concept.hbs (hunk 1 above)
Comment (andreo141) [reply app/templates/road-sign-concepts/road-sign-concept.hbs:113]: Yes noticed that yesterday aswell so added a '-' when it is empty like on the shape manager

File: app/models/road-sign-concept.ts [hunk 3]
```diff
@@ -16,11 +16,18 @@ import RoadSignCategory from './road-sign-category';
...
 import type { Type } from '@warp-drive/core-types/symbols';
+import type Variable from './variable';
 
 export default class RoadSignConcept extends TrafficSignConcept {
   //@ts-expect-error TS doesn't allow subclasses to redefine concrete types. We should try to remove the inheritance chain.
   declare [Type]: 'road-sign-concept';
 
+  @hasMany('variable', {
+    inverse: null,
+    async: true,
```
[the remark this answers is not in the document]
Comment (Windvis) [reply app/models/road-sign-concept.ts:27]: As discussed, you can ignore this.

**Comment by andreo141** [comment]
Linter is complaining about using the `did-insert` helper in my component. No idea why it complains on my component while it is being used throughout other parts of the codebase. @Windvis

**Comment by andreo141** [comment]
> Linter is complaining about using the `did-insert` helper in my component. No idea why it complains on my component while it is being used throughout other parts of the codebase. @Windvis

adding `'no-at-ember-render-modifiers': false,` to the `.template-lintrc.js` could be a quick fix, but idk why it only complains on the variable manager and not on other parts of the code.

**Comment by Windvis** [comment]
> Linter is complaining about using the `did-insert` helper in my component. No idea why it complains on my component while it is being used throughout other parts of the codebase. @Windvis

I updated the linting setup on the feature/v2 branch and this rule was part of that. The other places have template-lint-disable comments which is why they don't fail.

But I would just remove the `{{did-insert}}` and call the method from the constructor instead. I have no idea why `{{did-insert}}` is used so much in this app.

**Comment by andreo141** [comment]
> > Linter is complaining about using the `did-insert` helper in my component. No idea why it complains on my component while it is being used throughout other parts of the codebase. @Windvis
> 
> I updated the linting setup on the feature/v2 branch and this rule was part of that. The other places have template-lint-disable comments which is why they don't fail.
> 
> But I would just remove the `{{did-insert}}` and call the method from the constructor instead. I have no idea why `{{did-insert}}` is used so much in this app.

That's initially what i wanted to do. But after searching online i found out that it is bad practice to do async tasks inside of the constructor. So using `did-insert` seemed like the most logical thing to do to fetch data when a component is rendered.

File: app/components/road-sign-form/shape-manager.ts [hunk 4]
```diff
@@ -1,7 +1,7 @@
 import { service } from '@ember/service';
 import Component from '@glimmer/component';
 import { tracked } from '@glimmer/tracking';
-import type Store from 'ember-data/store';
```
Comment (Windvis) [inline app/components/road-sign-form/shape-manager.ts:4]: can you revert this again :smile:

Comment (andreo141) [reply app/components/road-sign-form/shape-manager.ts:4]: This actually was a lint error: `Imports from @ember-data packages should be preferred over imports from ember-dataeslint[ember/use-ember-data-rfc-395-imports](https://github.com/ember-cli/eslint-plugin-ember/tree/master/docs/rules/use-ember-data-rfc-395-imports.md)`

Comment (Windvis) [reply app/components/road-sign-form/shape-manager.ts:4]: Ah, ok, I thought this was your temporary commit :smile:

In any case, that rule is outdated, I think we ignore it in some other places. I fixed the issue in the linter package, but we need to update the package to get that fix (which we should probably do in a separate PR).

https://github.com/ember-cli/eslint-plugin-ember/blob/master/CHANGELOG.md#bug-bug-fix

**Review by Windvis — APPROVED** [review]

**Pull request merged**